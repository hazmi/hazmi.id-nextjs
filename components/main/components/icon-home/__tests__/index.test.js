import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'
import IconHome from '../index'

describe('With Enzyme', () => {
  it('Show the correct default svg', () => {
    const wrapper = shallow(<IconHome />)
    expect(wrapper.find('path').prop('fill')).toEqual('#444')
  })
  it('Show the correct custom color', () => {
    const wrapper = shallow(<IconHome fill="red" />)
    expect(wrapper.find('path').prop('fill')).toEqual('red')
  })
})

describe('With Snapshot Testing', () => {
  it('Show the correct default svg', () => {
    const component = renderer.create(<IconHome />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Show the correct custom color', () => {
    const component = renderer.create(<IconHome fill="red" />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})