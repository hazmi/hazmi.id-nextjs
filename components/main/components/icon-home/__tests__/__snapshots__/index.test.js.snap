// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Show the correct custom color 1`] = `
<svg
  fill="none"
  height={24}
  width={24}
>
  <path
    clipRule="evenodd"
    d="M4.889 8l6.001-6.001L16.891 8h-.001v10h-12V8h-.001zM2.89 9.999l-1.477 1.476L0 10.063 9.477.585c.78-.78 2.046-.78 2.826 0l9.477 9.477-1.413 1.414-1.477-1.477V18a2 2 0 0 1-2 2h-12a2 2 0 0 1-2-2V9.999z"
    fill="red"
    fillRule="evenodd"
  />
</svg>
`;

exports[`With Snapshot Testing Show the correct default svg 1`] = `
<svg
  fill="none"
  height={24}
  width={24}
>
  <path
    clipRule="evenodd"
    d="M4.889 8l6.001-6.001L16.891 8h-.001v10h-12V8h-.001zM2.89 9.999l-1.477 1.476L0 10.063 9.477.585c.78-.78 2.046-.78 2.826 0l9.477 9.477-1.413 1.414-1.477-1.477V18a2 2 0 0 1-2 2h-12a2 2 0 0 1-2-2V9.999z"
    fill="#444"
    fillRule="evenodd"
  />
</svg>
`;
